// angular-truncate-pipes/bundles/angular-truncate-pipes.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core'], factory) :
	(factory((global.ng = global.ng || {}, global.ng.angularTruncatePipes = global.ng.angularTruncatePipes || {}),global.ng.core));
}(this, (function (exports,_angular_core) { 'use strict';

var WordsPipe = (function () {
    function WordsPipe() {
    }
    /**
     * @param {?} value
     * @param {?=} args
     * @return {?}
     */
    WordsPipe.prototype.transform = function (value, args) {
        if (args === void 0) { args = 4; }
        var /** @type {?} */ res = value;
        if (value && args > 0) {
            var /** @type {?} */ valueWords = value.split(/\s+/);
            if (valueWords.length > args) {
                res = valueWords.slice(0, args).join(' ') + '…';
            }
        }
        return res;
    };
    return WordsPipe;
}());
WordsPipe.decorators = [
    { type: _angular_core.Pipe, args: [{
                name: 'words'
            },] },
];
/**
 * @nocollapse
 */
WordsPipe.ctorParameters = function () { return []; };
var SplitcharactersPipe = (function () {
    function SplitcharactersPipe() {
    }
    /**
     * @param {?} value
     * @param {?=} args
     * @return {?}
     */
    SplitcharactersPipe.prototype.transform = function (value, args) {
        if (args === void 0) { args = 10; }
        var /** @type {?} */ res = value;
        if (value && value.length > args && args > 0) {
            var /** @type {?} */ prefix = value.substring(0, args / 2);
            var /** @type {?} */ postfix = value.substring(value.length - args / 2, value.length);
            res = prefix + "..." + postfix;
        }
        return res;
    };
    return SplitcharactersPipe;
}());
SplitcharactersPipe.decorators = [
    { type: _angular_core.Pipe, args: [{
                name: 'splitcharacters'
            },] },
];
/**
 * @nocollapse
 */
SplitcharactersPipe.ctorParameters = function () { return []; };
var CharactersPipe = (function () {
    function CharactersPipe() {
    }
    /**
     * @param {?} value
     * @param {?=} args
     * @param {?=} breakOnWord
     * @return {?}
     */
    CharactersPipe.prototype.transform = function (value, args, breakOnWord) {
        if (args === void 0) { args = 10; }
        var /** @type {?} */ res = value;
        if (value && value.length > args && args > 0) {
            value = value.substring(0, args);
            if (breakOnWord) {
                while (value.charAt(value.length - 1) === ' ') {
                    value = value.substr(0, value.length - 1);
                }
            }
            else {
                var /** @type {?} */ lastspace = value.lastIndexOf(' ');
                // get last space
                if (lastspace !== -1) {
                    value = value.substr(0, lastspace);
                }
            }
            res = value + "...";
        }
        return res;
    };
    return CharactersPipe;
}());
CharactersPipe.decorators = [
    { type: _angular_core.Pipe, args: [{
                name: 'characters'
            },] },
];
/**
 * @nocollapse
 */
CharactersPipe.ctorParameters = function () { return []; };
var TruncatePipesModule = (function () {
    function TruncatePipesModule() {
    }
    /**
     * Use in AppModule
     * @return {?}
     */
    TruncatePipesModule.forRoot = function () {
        return {
            ngModule: TruncatePipesModule,
            providers: []
        };
    };
    /**
     * Use in features modules with lazy loading
     * @return {?}
     */
    TruncatePipesModule.forChild = function () {
        return {
            ngModule: TruncatePipesModule,
            providers: []
        };
    };
    return TruncatePipesModule;
}());
TruncatePipesModule.decorators = [
    { type: _angular_core.NgModule, args: [{
                declarations: [
                    CharactersPipe,
                    SplitcharactersPipe,
                    WordsPipe
                    // Pipes.
                    // Directives.
                ],
                exports: [
                    CharactersPipe,
                    SplitcharactersPipe,
                    WordsPipe
                    // Pipes.
                    // Directives.
                ]
            },] },
];
/**
 * @nocollapse
 */
TruncatePipesModule.ctorParameters = function () { return []; };

exports.TruncatePipesModule = TruncatePipesModule;
exports.ɵa = CharactersPipe;
exports.ɵb = SplitcharactersPipe;
exports.ɵc = WordsPipe;

Object.defineProperty(exports, '__esModule', { value: true });

})));
