// @ng-select/ng-select/esm2015/ng-select.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
/**
 * Generated bundle index. Do not edit.
 */
export { NgSelectComponent, NG_SELECT_DEFAULT_CONFIG, NgSelectModule } from './index';
export { ConsoleService as ɵo } from './ng-select/console.service';
export { NgDropdownPanelComponent as ɵk } from './ng-select/ng-dropdown-panel.component';
export { NgOptionHighlightDirective as ɵp } from './ng-select/ng-option-highlight.directive';
export { NgOptionComponent as ɵn } from './ng-select/ng-option.component';
export { NgFooterTemplateDirective as ɵf, NgHeaderTemplateDirective as ɵe, NgLabelTemplateDirective as ɵc, NgLoadingTextTemplateDirective as ɵi, NgMultiLabelTemplateDirective as ɵd, NgNotFoundTemplateDirective as ɵg, NgOptgroupTemplateDirective as ɵb, NgOptionTemplateDirective as ɵa, NgTagTemplateDirective as ɵj, NgTypeToSearchTemplateDirective as ɵh } from './ng-select/ng-templates.directive';
export { VirtualScrollService as ɵl } from './ng-select/virtual-scroll.service';
export { WindowService as ɵm } from './ng-select/window.service';
